app/views/devise/mailer/reset_password_instructions.html.erb

```diff
@@ -1,127 +1,28 @@
-        width: auto;
-      }
-
-      .title img {
-        padding: 20px;
-      }
-
-      .title h2 {
-        float: right;
-        padding: 20px;
-        margin: 0;
-      }
-
-      .content {
-        background-color: #FEFEFE;
-        width: auto;
-        padding: 40px 60px 60px 60px;
-      }
-
-      .footer {
-        text-align: center;
-        width: auto;
-        font-size: 0.8em;
-        color: #AAA;
-        margin-top: 20px;
-        margin-bottom: 40px;
-      }
-
-      @media only screen and (max-width: 640px) {
-        body {
-          background-color: #FEFEFE;
-        }
-        .title {
-          text-align: center !important;
-          width: auto;
-        }
-        .title h2 {
-          padding-top: 0;
-          float: none;
-        }
-        .container {
-          width: 100% !important;
-          -webkit-box-shadow: none;
-          box-shadow: none;
-        }
-        .content {
-          padding: 20px;
-        }
-      }
-
-      </style>
-  </head>
-  <body>
-    <div class="container">
-      <div class="title">
-        <%= image_tag('swapsea-logo-col-wht.png', width: "90%", alt: "Swapsea", class: "logo") %>
-        <h2>Choose a Password</h2>
-      </div>
-      <div class="content">
-        <p>
-          G'day <%= @resource.first_name if !@resource.has_multiple_emails? %>,
-        </p>
-        <p>
-          Almost there!
-        </p>
-        <p>
-			    Please <%= link_to 'choose a password', edit_password_url(@resource, :reset_password_token => @token) %> for Swapsea.
-		</p>
-		<p>If you don't want reset your password, please ignore this email.</p>
-  	<p>
-			<strong>'Invalid Token' error</strong><br/>
-			The link above expires after 24 hours. If you are getting an 'Invalid Token' error, please try password reset again.
-		</p>
-		<p>
-    <strong>What is Swapsea?</strong><br/>
-          Swapsea is an online swap system for Surf Life Saving Clubs that allows you to swap your patrols with other club members. Swapsea will also remind you of upcoming patrols so you don't forget. Simply login and view your roster on the dashboard.
-        </p>
-		<p>Much love,<br/>
-		Swapsea</p>
-      </div>
-    </div>
-    <div class="footer">This email was sent to <%= @resource.email %> after a request from www.swapsea.com.au. If you received this message in error, please contact <%= @user.club.name %>, or help@swapsea.com.au.<br/>
-      &copy; <%= Date.current.year %> Swapsea <br/>
-    </div>
-  </body>
-</html>
+<p>
+  G'day <%= @user.first_name %>,
+</p>
+<p>
+  Almost there!
+</p>
+<p>
+  <%= link_to 'Choose a password' , edit_password_url(@resource, :reset_password_token=> @token) %> for Swapsea.
+</p>
+<p>If you don't want to reset your password, please ignore this email.</p>
+<p>
+  <strong>'Invalid Token' error</strong><br />
+  The link above expires after 24 hours. If you are getting an 'Invalid Token' error, please <%=
+    link_to 'try password reset again' , activate_url %>.
```

Comment:
Is `activate_url` the right URL for generating a new password reset token? Should it be `new_user_password_url`?

Reply:
Good question - and you can probably explain this to me!
They look like two separate routes, yet they both trigger the same email `reset_password_instructions`.

I think what happens is this email is triggered both for Password Reset _and_ Activation.
We do Activation to confirm the member's email is still valid. Once this is done (once per season) they rarely reset their password, so the email content is focussed on one-time password set.

I think the Activate URL goes to the page that prompts for an email to generate a new short-lived activation token (for password set). Whereas the `edit_password_url` is the short-lived URL that does set the password .

HTH

Reply:
Yes, in this context it may be better to use `new_user_password_url`. From memory, the `activate_url` uses a custom route and controller action to allow users to "activate" accounts that had been pulled in from Surfguard. This was simply to show a view that displayed "activation" rather than  "forgot password" messaging to new users and avoid confusion. The underlying logic is based on Devise's forgot password workflow, as the mechanics are the same, so would deliver the same result.